5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+      type: string
+      pattern: ^[A-Za-z0-9-_]*={0,2}$
+      example: MWtzOXIpaUEsOyEhc1NsTwo=
+
+    QueryExpression:
+      description: |
+        An expression of facts about a desired set of service records. A query expression consists
+        of three parts, a _service part_, a _provider part_ and a _limit part_. At least one of the 
+        two former parts must be part of the query, or it will always match zero records.
+
+        The __service part__ must at least name a specific service type, but may also include a
+        version specifier. If given, the version specifier will match records with any other version
+        that is _compatible_ with it. A version specifier consist of a MAJOR, MINOR and a PATCH
+        number, separated by dots. If both or the latter of the MINOR and PATCH numbers are omitted
+        from a given version, the ones omitted are assumed to be zero. A given version is compatible
+        with a _candidate_ version if their MAJOR numbers are equal and the MINOR number of the
+        candidate version is greater than that of the given version. They are also compatible if
+        their MAJOR and MINOR numbers are equal and the PATCH number of the candidate version is
+        equal to or greater than that of the given version. For example, the query expression
+        `my-service~1.4` matches `my-service~1.4.1` and `my-service~1.6.5`. But it does not match
+        `my-service~1.3.9` or `my-service~2.0.4`.
+
+        The __provider part__ limits matches to records that refer to the identified service
+        provider.
+
+        The __limit part__ sets a limit on how many matching records to include in the response. It
+        consists of an asterisk followed by a positive integer, as in `a-service~1.0*4`.
+      type: string
+      pattern: ^(?:(?<type>[^!*,:@^~]+)(?:~(?<version>\d+(?:\.\d+(?:\.\d+)?)?))?)?(?:@(?<provider>[A-Za-z0-9-_]*={0,2}))?(?:\*(?<limit>\d+))?$
+      example: se.arrowhead.service-discovery-http-json~1.6@MWtzOXIpaUEsOyEhc1NsTwo=
+
+    Service:
+      type: object
+      properties:
+        name: 
+          type: string
+          description: Service instance human-readable name
+        service-type:
+          $ref: '#/components/schemas/TypeIdentifier'
+        version: 
+          $ref: '#/components/schemas/Version'
+        addresses:
+          description:
+            A collection of addresses, each with a unique tag name. The tag names _may_ be used to
+            help a service consumer chose the most appropriate address. The service consumer may,
+            however, simply try all of them in any order until it finds one that works.
+          type: object
+          patternProperties:
+            .*: { $ref: '#/components/schemas/Address' }
+        interface:
+          type: array
+          items:
+            $ref: '#/components/schemas/Interface'
+        auth-info: 
```

Comment:
In my opinion, this is part of the API specification and thus should not be included here. Instead, this is implicit from the consumer implementing the specification.